cloudinit/sources/DataSourceCloudStack.py

```diff
@@ -208,38 +211,43 @@ def _get_data(self):
             LOG.debug("Using seeded cloudstack data from: %s", self.seed_dir)
             return True
         try:
+            if self.perform_dhcp_setup:
+                LOG.debug("Setting up network to reach IMDS")
+                candidate_nics = net.find_candidate_nics()
+                LOG.debug("Looking for the primary NIC in: %s", candidate_nics)
+
+                for candidate_nic in candidate_nics:
+                    network_context = EphemeralIPNetwork(
+                        self.distro,
+                        candidate_nic,
+                    )
+                    try:
+                        with network_context:
```

Comment:
No need for the network_context local I don't think, let's consolidate.
```suggestion
                    try:
                        with EphemeralIPNetwork(self.distro, candidate_nic):
```